src/sinks/aws_s_s/sqs/client.rs

```diff
@@ -0,0 +1,41 @@
+use super::{Client, SendMessageEntry, SendMessageResponse};
+use aws_sdk_sqs::{error::SendMessageError, types::SdkError};
+use futures::TryFutureExt;
+use tracing::Instrument;
+
+#[derive(Clone, Debug)]
+pub struct SqsMessagePublisher {
+    client: aws_sdk_sqs::Client,
+    queue_url: String,
+}
+
+impl SqsMessagePublisher {
+    pub fn new(client: aws_sdk_sqs::Client, queue_url: String) -> Self {
+        Self { client, queue_url }
+    }
+}
+#[async_trait::async_trait]
+impl Client for SqsMessagePublisher {
+    async fn send_message(
+        &self,
+        entry: SendMessageEntry,
+        byte_size: usize,
+    ) -> Result<SendMessageResponse, SdkError<SendMessageError>> {
```

Comment:
@neuronull Do you have any idea how to abstract the error from this trait? I tried with different variations and I either run into some PhantomData problem (unused generic params) or some variation of generic associated types.  Both are somewhat challenging for someone doing Rust since 6 hours 🙈

Reply:
Is this still relevant? It looks like you overcame this in https://github.com/vectordotdev/vector/pull/18141/commits/891e119e235370b396207629702b0ca1c572eaa5

Reply:
Yep, but if you know any prettier or better suggestion I'd be happy to hear them